src/main/ruby/core/type.rb

```diff
@@ -671,6 +671,9 @@ def self.check_safe_level(level)
       if level > 1 # SAFE_LEVEL_MAX
         raise ArgumentError, '$SAFE=2 to 4 are obsolete'
       end
+      if level != 0
+        warn '[ruby] WARN $SAFE level checks not implemented' if $VERBOSE
```

Comment:
Can you use `Truffle::Debug.log_warning` instead.

Reply:
Yes, updated to `log_warning`.